Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,248 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+
+Providing this capability is optional for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string | https callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentication token for callback API endpoint. It MUST be indicated within HTTP Authorization header as a Bearer Token format, e.g. ```Authorization: Bearer $notificationAuthToken``` | optional |
+
+Format conventions regarding ```notificationAuthToken``` attribute, in order to provide Uniqueness, Randomness and Simplicity for its management are the following:
+- It MUST BE an opaque attribute
+- It MUST NOT be a JWT Token
+- It has to be a restricted length, being an string between [20-64] characters.
+- It is HIGHLY recommended to have random-based pattern, like UUIDv4 format (32 hexadecimal characters, separated by `-`, e.g. ```550e8400-e29b-41d4-a716-446655440000```) 
+
+**Resource-based subscription**
+
+A resource-based subscription is a subscription managed as a resource. An endpoint is provided to request subscription creation.  As this subscription is managed as an API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several event types managed. The subscription endpoint will be unique but 'eventType' attribute is used to distinguish distinct events subscribed.
+
+In order to ease developer adoption, the pattern for Resource-based subscription should be consistent for all API providing this feature.
+
+4 operations must be defined:
+
+| operation | path | description |
+| ----- |	-----  |	 -----  | 
+| POST | `/subscriptions` |  Operation to request a subscription.     |
+| GET | `/subscriptions` |  Operation to retrieve a list of subscriptions - could be an empty list.  eg. `GET /subscriptions?type=ROAMING_STATUS&ExpireTime.lt=2023-03-17` |
+| GET | `/subscriptions/{subscriptionId}` | Operation to retrieve a subscription |
+| DELETE | `/subscriptions/{subscriptionId}` | Operation to delete a subscription |
+
+
+Note on operation path:
+The recommended pattern is to use `/subscription` path for the subscription operation. But API design team, for specific case, has the option to append `/subscription` path with a prefix (e.g. `/roaming/subscription` and `/connectivity/subscription`). The rationale for using this alternate pattern should be explicitly provided (e.g. the notification source for each of the supported events may be completely different, in which case separating the implementations is beneficial). 
```

Comment:
OK, even I am going to comment in the thread

Reply:
typo in the recommended pattern statement - "subscription" should perhaps be pluralized to be consistent.